tmt/trying.py

```diff
@@ -0,0 +1,315 @@
+""" Easily try tests and experiment with guests """
+
+import textwrap
+from typing import Any
+
+import click
+import fmf
+from fmf.utils import listed
+
+import tmt
+import tmt.log
+import tmt.steps
+import tmt.steps.provision
+import tmt.utils
+from tmt import Plan
+from tmt.templates import DEFAULT_PLAN, DEFAULT_PLAN_NAME
+from tmt.utils import MetadataError, Path
+
+USER_PLAN_NAME = "/user/plan"
+
+
+class Try(tmt.utils.Common):
+
+    def __init__(
+            self,
+            *,
+            tree: tmt.Tree,
+            logger: tmt.log.Logger,
+            **kwargs: Any) -> None:
+        """ Just store the tree """
+
+        super().__init__(logger=logger, **kwargs)
+
+        self.tree = tree
+        self.tests: list[tmt.Test] = []
+        self.plans: list[Plan] = []
+        self.image_and_how = self.opt("image_and_how")
+
+    def check_tree(self) -> None:
+        """ Make sure there is a sane metadata tree """
+
+        # Both tree and root should be defined
+        try:
+            if self.tree and self.tree.root:
+                return
+
+        # Create a dumb fmf Tree if no metadata around
+        except MetadataError:
+            new_fmf_tree = fmf.Tree({"nothing": "here"})
+            self.tree.tree = new_fmf_tree
+
+    def check_tests(self) -> None:
+        """ Check for available tests """
+
+        # Search for tests according to provided names
+        test_names = list(self.opt("test"))
+        if test_names:
+            self.tests = self.tree.tests(names=test_names)
+            if not self.tests:
+                raise tmt.utils.GeneralError(
+                    f"No test matching '{listed(test_names)}' found.")
+
+        # Default to tests under the current working directory
+        else:
+            if not self.tree.root:
```

Comment:
always false, you already checked this in check_tree and created a dummy tree, right?

Reply:
Actually, the dummy `fmf.Tree` created from a `dict` does not have `root` set. Plus, `self.tree.root` a `tmt.Tree.root` is a different from `self.tree.tree.root` which is `fmf.Tree.root`. Each `tmt.Tree` keeps reference to the origin `fmf.Tree` from which it was created. Entering an empty directory triggers this:

```
> mkdir /tmp/nothing
> cd /tmp/nothing
> tmt -d try
/var/tmp/tmt/run-014
Use the default user plan config.
Workdir '/var/tmp/tmt/run-014/user/plan/local' created.
Matching plans found
/user/plan/local
No fmf tree root, no tests.
Let's try something with /user/plan/local.
```